flycheck-elsa.el

```diff
@@ -46,23 +45,65 @@
   :group 'flycheck-elsa
   :type '(repeat regexp))
 
+(defun flycheck-elsa--locate-cask-dir ()
+  "Return dir located Cask file.  If missing, return nil."
+  (when-let (file (locate-dominating-file (buffer-file-name) "Cask"))
+    (file-name-directory file)))
+
+(defun flycheck-elsa--elsa-dependency-p ()
```

Comment:
This will break if cask changes the format of their files.  That's why we are using their API to read the `Cask` file.  There should be a way to read it without side-effect.

Reply:
I can't. My proposal is to remove cask dependency fixes any issues so I can't use cask API.
Cask is already used in many packages, and if additional grammars will be added, but are expected to change with compatibility.

Even in situations where the Cask API cannot be used, decoding a Cask file in this way is easy and will take a less day if there is a format change.